bolt-simple-taproot.md

```diff
@@ -377,348 +352,276 @@ The Context column decodes as follows:
-  - MUST validate that the 
-  - MUST validate the signature as a _partial_ signature based on the
-    `PartialSigVerifyInternal` (using the verification nonce of the local
-    party) algorithm of `bip-musig2`:
 
-    - if the partial signature is invalid, MUST fail the channel
+- MUST fail the channel if `partial_signature_with_nonce` is absent.
+- MUST fail the channel if `musig2_nonce` is absent.
+- MUST compute the aggregate nonce from:
+  - the `musig2_nonce` field the recipient previously sent in `accept_channel`
+  - the `public_nonce` included as part of the `partial_signature_with_nonce` field
+- MUST verify the `partial_signature_with_nonce` field using the `PartialSigVerifyInternal`
+  algorithm of `bip-musig2`:
+  - if the partial signature is invalid, MUST fail the channel
+- MUST store `musig2_nonce` until they send the `funding_signed` message.
+  - SHOULD only store `musig2_nonce` in RAM, and not persist it to disk.
 
 #### `funding_signed` Extensions
 
-The sender MUST set:
+1. `tlv_stream`: `funding_signed_tlvs`
+2. types:
+   1. type: 2 (`partial_signature_with_nonce`)
+   2. data:
+      * [`98*byte`: `partial_signature || public_nonce`]
 
-  - `signature` to the valid signature, using its `funding_pubkey` for the
-    initial commitment transaction, as defined in this documnt.
-    - this signature MUST be a partial signature as defined by `bip-musig2`.
+##### Requirements
 
-The recipient:
+The sender:
 
-  - MUST validate the signature as a _partial_ signature based on the
-    `PartialSigVerifyInternal` algorithm of `bip-musig2`:
+- MUST sort the exchanged `funding_pubkey`s using the `KeySort` algorithm from `bip-musig2`.
+- MUST compute the aggregated `musig2` public key from the sorted `funding_pubkey`s using the `KeyAgg` algorithm
+  from `bip-musig2`.
+- MUST generate a unique nonce to combine with the `musig2_nonce` previously received in
+  `funding_created` using `NonceAgg` from `bip-musig2`.
+- MUST use the generated secret nonce and the calculated aggregate nonce to construct a
+  `musig2` partial signature for the sender's remote commitment using the `Sign` algorithm from `bip-musig2`.
+- MUST include the partial signature and the public counterpart of the generated nonce in
+  the `partial_signature_with_nonce` field.
 
-    - if the partial signature is invalid, MUST fail the channel
+The recipient:
 
-#### `funding_locked` Extensions
+- MUST fail the channel if `partial_signature_with_nonce` is absent.
+- MUST compute the aggregate nonce from:
+  - the `musig2_nonce` field the recipient previously sent in `funding_created`
+  - the `public_nonce` included as part of the `partial_signature_with_nonce` field
+- MUST verify the `partial_signature_with_nonce` field using the `PartialSigVerifyInternal`
+  algorithm of `bip-musig2`:
+  - if the partial signature is invalid, MUST fail the channel
 
-We add a new TLV field to the `funding_locked` message:
+#### `channel_ready`/`channel_reestablish` Extensions
```

Comment:
The difference is that for channel_read/funding_locked there's no retransmission, as no state updates have happened yet. For chan_reest, both sides _must_ regenerate their signatures using the newly exchanged nonces (gen new signing nonce, use received verification nonce of the remote party). 

It's worth pointing out this difference in chan_reest, as today, to retransmit a sig you just need to read it from disk and send it over. With these changes, you _must_ regenerate it given the new nonce, since we don't want to force either side to write to disk their unused nonces (otherwise I'd need to remember which nonce I sent you before the D/C to be able to make the final sig).

Reply:
I am not entirely sure I agree with your last point. If following a disconnect I need to make the final sig, I have no choice but to either remember the last (remote) nonce you sent, or to at the very least remember the last aggregate nonce I calculated with the remote nonce you sent. So that requirement is there either way, no? It's either that or storing the full signature.

Reply:
We split `channel_ready` and `chanenl_reestablish` back up again, and incorporated the commitment signature retransmission requirements in the latter section.